sdk/lib/drivers/arbiter/arbiter.c

```diff
@@ -34,10 +230,135 @@ ArbInitializeArbiterInstance(
+        ExFreePoolWithTag(Arbiter->AllocationStack, 'AbrA');
+        ExFreePoolWithTag(Arbiter->MutexEvent, 'MbrA');
+        return STATUS_INSUFFICIENT_RESOURCES;
+    }
+
+    Arbiter->PossibleAllocation = ExAllocatePoolWithTag(PagedPool, sizeof(RTL_RANGE_LIST), 'RbrA');
+    if (!Arbiter->PossibleAllocation)
+    {
+        DPRINT1("ArbInitializeArbiterInstance: STATUS_INSUFFICIENT_RESOURCES\n");
+        ExFreePoolWithTag(Arbiter->Allocation, 'RbrA');
+        ExFreePoolWithTag(Arbiter->AllocationStack, 'AbrA');
+        ExFreePoolWithTag(Arbiter->MutexEvent, 'MbrA');
+        return STATUS_INSUFFICIENT_RESOURCES;
+    }
+
+    RtlInitializeRangeList(Arbiter->Allocation);
+    RtlInitializeRangeList(Arbiter->PossibleAllocation);
+
+    Arbiter->Name = ArbiterName;
+    Arbiter->ResourceType = ResourceType;
+    Arbiter->TransactionInProgress = FALSE;
+
+    if (!Arbiter->TestAllocation)
+        Arbiter->TestAllocation = ArbTestAllocation;
+
+    if (!Arbiter->RetestAllocation)
+        Arbiter->RetestAllocation = ArbRetestAllocation;
+
+    if (!Arbiter->CommitAllocation)
+        Arbiter->CommitAllocation = ArbCommitAllocation;
+
+    if (!Arbiter->RollbackAllocation)
+        Arbiter->RollbackAllocation = ArbRollbackAllocation;
+
+    if (!Arbiter->AddReserved)
+        Arbiter->AddReserved = ArbAddReserved;
+
+    if (!Arbiter->PreprocessEntry)
+        Arbiter->PreprocessEntry = ArbPreprocessEntry;
+
+    if (!Arbiter->AllocateEntry)
+        Arbiter->AllocateEntry = ArbAllocateEntry;
+
+    if (!Arbiter->GetNextAllocationRange)
+        Arbiter->GetNextAllocationRange = ArbGetNextAllocationRange;
+
+    if (!Arbiter->FindSuitableRange)
+        Arbiter->FindSuitableRange = ArbFindSuitableRange;
+
+    if (!Arbiter->AddAllocation)
+        Arbiter->AddAllocation = ArbAddAllocation;
+
+    if (!Arbiter->BacktrackAllocation)
+        Arbiter->BacktrackAllocation = ArbBacktrackAllocation;
+
+    if (!Arbiter->OverrideConflict)
+        Arbiter->OverrideConflict = ArbOverrideConflict;
+
+    if (!Arbiter->BootAllocation)
+        Arbiter->BootAllocation = ArbBootAllocation;
+
+    if (!Arbiter->QueryConflict)
+        Arbiter->QueryConflict = ArbQueryConflict;
+
+    if (!Arbiter->StartArbiter)
+        Arbiter->StartArbiter = ArbStartArbiter;
+
+    Status = ArbBuildAssignmentOrdering(Arbiter, OrderName, OrderName, TranslateOrderingFunction);
+    if (NT_SUCCESS(Status))
+    {
+        return STATUS_SUCCESS;
+    }
+
+    DPRINT1("ArbInitializeArbiterInstance: Status %X\n", Status);
+
+    if (Arbiter->MutexEvent)
+    {
+        ExFreePoolWithTag(Arbiter->MutexEvent, 'MbrA');
+    }
+    if (Arbiter->Allocation)
+    {
+        ExFreePoolWithTag(Arbiter->Allocation, 'RbrA');
+    }
+    if (Arbiter->PossibleAllocation)
+    {
+        ExFreePoolWithTag(Arbiter->PossibleAllocation, 'RbrA');
+    }
+    if (Arbiter->AllocationStack)
+    {
+        ExFreePoolWithTag(Arbiter->AllocationStack, 'AbrA');
+    }
```

Comment:
No need to check pointers here. If we reached this, all of them are initialized

Reply:
Ok